shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -195,28 +198,128 @@ public boolean sendKeyEvent(KeyEvent event) {
                 }
             } else if (event.getKeyCode() == KeyEvent.KEYCODE_DPAD_LEFT) {
                 int selStart = Selection.getSelectionStart(mEditable);
-                int newSel = Math.max(selStart - 1, 0);
-                setSelection(newSel, newSel);
+                int selEnd = Selection.getSelectionEnd(mEditable);
+                if (selStart == selEnd && !event.isShiftPressed()) {
+                    int newSel = Math.max(selStart - 1, 0);
+                    setSelection(newSel, newSel);
+                }
+                else {
```

Comment:
Nit: I think the usual style in our Java files is to put this on the same line like `} else {`.  Also below.